src/com/gfredericks/catch_data.clj

```diff
@@ -10,6 +10,12 @@
   [x]
   (and (seq? x) (= 'finally (first x))))
 
+(defn ^:private apply-predicate
+  [predicate data-sym]
+  (if (map? predicate)
+    `(= (select-keys ~data-sym (keys ~predicate)) ~predicate)
```

Comment:
have to be careful with unquoting the same thing twice; would be better to wrap in a `(let [predicate# ~predicate] ...)`, otherwise you'll get things evaluated twice which could be weird if the map isn't strictly data.